hardware/gpsPi/gps_reader.py

```diff
@@ -0,0 +1,59 @@
+import os
+import serial
+from ..Utils.utils import get_logger
+from hardware.Utils.utils import date_str_with_current_timezone
+
+GPS_ID = 10
+
+
+class GPSReader:
+    def __init__(self, log_file_name=None):
+        self.gps = serial.Serial(os.environ["GPS_PORT"], os.environ["GPS_BAUDRATE"])
+
+        if log_file_name is None:
+            self.logging = get_logger("GPS_LOG_FILE")
+        else:
+            self.logging = get_logger(log_file_name, log_file_name)
+
+    def get_geolocation(self):
+        while self.gps.inWaiting() == 0:
```

Comment:
That is a good point. While I think it could possibly be made more robust, that use case is sorta handled by lines 57-61 in `hardware/main.py`. @PrabhanshuAttri  do you have anything to add?

Reply:
Yes, I agree with both of you. I don't have much idea about detecting GPS Hat. @MichaelLally @yashasvid Is there a way to implement this?

Reply:
@VentusXu09 
We can add this as a separate feature in a later update. I believe there is a flag in one of the GPS data streams that basically goes from 0 (good coordinates) to 4 (pretty much a guess). We could package this flag with the coordinate data as `values = {latitude, longitude, quality}` or something like that

I don't think it should be included on this PR though

Reply:
We could also just not send data if the flag is >1. We'll look into it

Reply:
Cool. I agree this should be implemented in the future.